### PR TITLE
fix: support multiple mappers with the same name in different namespaces

### LINKED ISSUE #122: Mappers with same type name on different namespaces
**Describe the bug**
Mapperly fails to generate code when two or more mapper classes have the same name.

**To Reproduce**
1. Declare a mapper with the name "CarMapper" on the namespace "Test.A"
2. Declare another mapper with the same name "CarMapper" on the namespace "Test.B"
3. Add mapping methods to both mappers

**Expected behavior**
Code should be generated for both mappers

**Code snippets**
A/CarMapper.cs
```
namespace Test.A;

[Mapper]
internal partial class CarMapper
{
    internal partial CarADto CarToDto(CarA car);
}
```

B/CarMapper.cs
```
namespace Test.B;

[Mapper]
internal partial class CarMapper
{
    internal partial CarBDto CarToDto(CarB car);
}
```

**Environment (please complete the following information):**
 - Mapperly Version: 2.3.3
 - .NET Version: NET 6.0
 - Target Framework: net6.0
 - OS: Windows 10

### PR DESCRIPTION
**Pull request by latonz**
Closes #122

### CODE REVIEW

File: src/Riok.Mapperly/MapperGenerator.cs [hunk 1]
```diff
@@ -74,7 +75,7 @@ private static void Execute(Compilation compilation, ImmutableArray<ClassDeclara
             var descriptor = builder.Build();
 
             ctx.AddSource(
-                descriptor.FileName,
+                uniqueNameBuilder.Build(mapperSymbol.Name) + ".g.cs",
```
Comment (CommonGuy) [inline src/Riok.Mapperly/MapperGenerator.cs:78]: I would extract `.g.cs` into a constant

File: test/Riok.Mapperly.Tests/_snapshots/MapperTest.SameMapperNameInMultipleNamespacesShouldWork.01FooBarMapper1.g.verified.cs [hunk 2]
```diff
@@ -0,0 +1,12 @@
+﻿//HintName: FooBarMapper1.g.cs
+#nullable enable
+namespace Test.B
+{
+    internal partial class FooBarMapper
```
Comment (CommonGuy) [inline test/Riok.Mapperly.Tests/_snapshots/MapperTest.SameMapperNameInMultipleNamespacesShouldWork.01FooBarMapper1.g.verified.cs:5]: The leading `01` in the file name is caused by the test framework, right?

**Review by CommonGuy — CHANGES_REQUESTED** [review]

File: src/Riok.Mapperly/MapperGenerator.cs (hunk 1 above)
Comment (latonz) [reply src/Riok.Mapperly/MapperGenerator.cs]: Improved.

File: test/Riok.Mapperly.Tests/_snapshots/MapperTest.SameMapperNameInMultipleNamespacesShouldWork.01FooBarMapper1.g.verified.cs (hunk 2 above)
Comment (latonz) [reply test/Riok.Mapperly.Tests/_snapshots/MapperTest.SameMapperNameInMultipleNamespacesShouldWork.01FooBarMapper1.g.verified.cs:5]: It is.

**Review by CommonGuy — APPROVED** [review]

**Pull request merged**